python-package/lightgbm/basic.py

```diff
@@ -408,21 +408,27 @@ def _choose_param_value(main_param_name: str, params: Dict[str, Any], default_va
     # avoid side effects on passed-in parameters
     params = deepcopy(params)
 
-    # find a value, and remove other aliases with .pop()
-    # prefer the value of 'main_param_name' if it exists, otherwise search the aliases
-    found_value = None
+    aliases = set(_ConfigAliases.get(main_param_name)) - {main_param_name}
```

Comment:
This is already `set`
https://github.com/microsoft/LightGBM/blob/2f5baa3d39efb518cd13a7932fe4d8602c36762f/python-package/lightgbm/basic.py#L369
```suggestion
    aliases = _ConfigAliases.get(main_param_name) - {main_param_name}
```

Reply:
ah right! I forgot, thank you very much for the reminder.